src/id3_variants_training/ConfusionMatrix.py

```diff
@@ -15,81 +15,75 @@ def __init__(self, file_path, local):
         predicted is X axis
 
         Args:
-            file_path (str): Path to json file that contains the variant ranges
-            local (bool): Flag to determine whether or not to read locally or from a server
+            api (LOCAL_API | GA4GH_API): API object that is used to interact with the virtual API
 
         Attributes:
-            api (API): API object that is used to interact with the virtual API
+            api (LOCAL_API | GA4GH_API): API object that is used to interact with the virtual API
             root_node (Node): Creates the root node of the tree to be added upon
 
             length (int): length of all the ancestries
-            ancestry_list (list): list of all ancestries without repetitions
             conf_matrix (list): the confusion matrix based on the ID3 classifier
             diagonal_sum (int): sum of the diagonals within the matrix
             total (int): total sum of all values in matrix
-        '''
-        super(ConfusionMatrix, self).__init__(file_path, local)
-
-        # modify API for conf_matrix
-        self.api = LOCAL_API(file_path, conf_matrix=True)
-
-        # initialize ID3 algorithm
-        subset = self.api.get_target_set()
-        self.root_node = ID3_Node('root', subset, True)
-        self.ID3(self.root_node)
+        """
+        super(ConfusionMatrix, self).__init__(api)
 
         # create conf_matrix and calculate useful attributes
         self.length = len(self.api.ancestry_list)
-        self.conf_matrix = [ [0 for x in range(0, self.length)] for y in range(0, self.length) ]
+        self.conf_matrix = [[0] * self.length for _ in range(self.length)]
 
         for variants, popu in zip(self.api.test_variant_list, self.api.test_popu_list):
             include_variants = [self.api.variant_name_list[idx] for idx, is_variant in enumerate(variants) if is_variant == 1]
 
             # Actual Result
-            y = self.api.ancestry_list.index( popu )
+            y = self.api.ancestry_list.index(popu)
             # Predicted Result
-            x = self.api.ancestry_list.index( self.predict(include_variants).most_common_ancestry )
+            x = self.api.ancestry_list.index(self.predict(include_variants).most_common_ancestry)
 
             self.conf_matrix[y][x] += 1
 
-        self.diagonal_sum = sum( [self.conf_matrix[i][i] for i in range(0, self.length)] )
-        self.total = sum( [sum(self.conf_matrix[i]) for i in range(0, self.length)] )
+        self.diagonal_sum = sum([self.conf_matrix[i][i] for i in range(self.length)])
+        self.total = sum([sum(self.conf_matrix[i]) for i in range(self.length)])
 
-    def accuracy(self):
-        '''
+    def get_accuracy_rate(self):
+        """
```

Comment:
I think accuracy (and precision, below) is a better name than precision rate; precision is already a rate.  Not sure about accuracy vs get accuracy, either is fine.